### PR TITLE
Ensure max_user_instances is set to a sane value at startup

### PR DESCRIPTION
**Pull request by hughsie**
Fixes the hard-to-debug https://github.com/fwupd/fwupd/issues/3790

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
On the logic, it's better to "_just work_" rather than fail in a "_good way_".

**Review by superm1 — COMMENTED** [review]
I don't like a daemon going and changing this just because it has the permissions to do so.  What if an admin intentionally lowered it and is closely tracking what apps are using it and pushing the threshold?

**Comment by hughsie** [comment]
> I don't like a daemon going and changing this just because it has the permissions to do so

Agreed, it didn't sit 100% right with me either. I've switched it to just dump a warning in the logs instead.

**Comment by travier** [comment]
We had a "similar" issue (https://github.com/coreos/fedora-coreos-tracker/issues/637) and asked for it to be more dynamic (https://bugzilla.redhat.com/show_bug.cgi?id=1888617) and that landed in 5.11.

File: src/fu-engine.c [hunk 1]
```diff
@@ -89,6 +89,7 @@
 #endif
 
 #define MINIMUM_BATTERY_PERCENTAGE_FALLBACK 10
+#define INOTIFY_MIN_USER_INSTANCES	    64
```
Comment (superm1) [inline src/fu-engine.c:92]: Is this really going to find anything? 64 is veryyyyy low. Given the commit linked I have an idea.

How about if you detect it's less than 1% of system ram just leave a warning in the logs.

Comment (hughsie) [reply src/fu-engine.c:92]: >64 is veryyyyy low

Any higher than that and my fwupd started up just fine. I'm hesitant to make this a percentage of the RAM as that gets into all kind of edge cases. How about we make it hardcoded to 8192/2, i.e. 4096?

Comment (superm1) [reply src/fu-engine.c:92]: >Any higher than that and my fwupd started up just fine

Right - this should only be an issue while system is under pressure.  Like if you started up vscode with the C++ extension and opened a kernel directory and then started fwupd.

>How about we make it hardcoded to 8192/2, i.e. 4096

I'm not sure I agree with this either though.  It's not going to be a deterministic failure.  What about instead moving it back to the failure path like my original idea?

Comment (hughsie) [reply src/fu-engine.c:92]: Like https://github.com/fwupd/fwupd/pull/4060 ?

Comment (superm1) [reply src/fu-engine.c:92]: Yup, I think that's a lot more useful.

**Pull request closed without merging**